### PR TITLE
MINIFICPP-2035 - NiFi flow json configuration format support

### PR DESCRIPTION
**Pull request by adamdebreceni**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/include/core/flow/FlowSchema.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+#include <vector>
+#include <string>
+
+namespace org::apache::nifi::minifi::core::flow {
+
+struct FlowSchema {
+  using Keys = std::vector<std::string>;
+
+  Keys FLOW_HEADER;
```
[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/include/core/flow/FlowSchema.h:29]: Const data members disable move. I believe it's better to default to mutable non-static data members.

[the remark this answers is not in the document]
Comment (lordgamez) [reply libminifi/include/core/flow/FlowSchema.h:29]: I see, in that case I would not use full caps for the member names.

File: extensions/http-curl/protocols/RESTSender.cpp [hunk 2]
```diff
@@ -162,6 +162,7 @@ C2Payload RESTSender::sendPayload(const std::string& url, const Direction direct
   if (payload.getOperation() == Operation::TRANSFER) {
     auto read = std::make_unique<utils::HTTPReadCallback>(std::numeric_limits<size_t>::max());
     client.setReadCallback(std::move(read));
+    client.setRequestHeader("Accept", "application/vnd.minifi-c2+json;version=1, text/yml");
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply extensions/http-curl/protocols/RESTSender.cpp:165]: I think the json format is the default now in the docker tests, only a few tests are set specifically to use yaml. So I think the nifi C2 tests are running with the json config now so I suppose that should be okay. Maybe 1 more test could be added to test both versions.

File: libminifi/include/core/flow/FlowSchema.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamdebreceni) [reply libminifi/include/core/flow/FlowSchema.h]: renamed members in [ba24d56](https://github.com/apache/nifi-minifi-cpp/pull/1494/commits/ba24d56be408fc96f566235717e8b175affd2959)

File: extensions/http-curl/protocols/RESTSender.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamdebreceni) [reply extensions/http-curl/protocols/RESTSender.cpp:165]: it works with the nifi c2 service as long as we also specify `text/yml` if we explicitly specify the `Accept` header, some tests were failing without it, so the current tests also test compatibility with nifi c2 service

**Review by lordgamez — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Pull request closed without merging**